executor/cook/progress.py

```diff
@@ -246,43 +227,59 @@ def __init__(self, task_id, config, stop_signal, task_completed_signal, send_pro
             Event that tracks task execution completion
         send_progress_message: function(message)
             The helper function used to send the progress message
+        counter: ProgressSequenceCounter
+            The sequence counter
         location: string
             The target location to read for progress messages
-        counter: ProgressSequenceCounter
-            The sequence counter."""
+        location_tag: string
+            A tag to identify the target location."""
+        self.location_tag = location_tag
+        self.progress_sample_interval_ms = config.progress_sample_interval_ms
         self.progress_complete_event = Event()
-        self.watcher = ProgressWatcher(location, counter, config.max_bytes_read_per_line, config.progress_regex_string,
-                                       stop_signal, task_completed_signal)
-        self.updater = ProgressUpdater(task_id, config.max_message_length, config.progress_sample_interval_ms,
-                                       send_progress_message)
+        self.watcher = ProgressWatcher(location, location_tag, counter, config.max_bytes_read_per_line,
+                                       config.progress_regex_string, stop_signal, task_completed_signal)
+        self.updater = ProgressUpdater(task_id, config.max_message_length, send_progress_message)
 
     def start(self):
         """Launches a thread that starts monitoring the progress location for progress messages."""
-        logging.info('Starting progress monitoring from {}'.format(self.watcher.progress_target_file()))
+        logging.info('Starting progress monitoring from [tag={}]'.format(self.location_tag))
         Thread(target=self.track_progress, args=()).start()
 
     def wait(self, timeout=None):
         """Waits for the progress tracker thread to run to completion."""
         self.progress_complete_event.wait(timeout=timeout)
-        progress_target_file = self.watcher.progress_target_file()
         if self.progress_complete_event.isSet():
-            logging.info('Progress monitoring from {} complete'.format(progress_target_file))
+            logging.info('Progress monitoring from complete [tag={}]'.format(self.location_tag))
         else:
-            logging.info('Progress monitoring from {} did not complete'.format(progress_target_file))
+            logging.info('Progress monitoring from did not complete [tag={}]'.format(self.location_tag))
 
     def track_progress(self):
         """Retrieves and sends progress updates using send_progress_update_fn.
+        It throttles the rate at which progress updates are sent.
         It sets the progress_complete_event before returning."""
         try:
+            last_reported_time = None
+
+            def has_enough_time_elapsed_since_last_update():
```

Comment:
Is this just lifting the check from the ProgressWatcher to the caller? What's the rationale for doing that?

Reply:
Yes, this moves the throttle from `ProgressUpdater` to `ProgressTracker` as a refactoring. I will move it to a separate PR if you prefer.

Reply:
Removed the changes from this PR